src/predicate/str/difference.rs

```diff
@@ -0,0 +1,69 @@
+use difference;
+
+use Predicate;
+
+/// Predicate that diffs two strings.
+///
+/// This is created by the `predicate::str::diff`.
+#[derive(Clone, Debug)]
+pub struct DifferencePredicate {
+    orig: String,
+    split: String,
+    distance: i32,
+}
+
+impl DifferencePredicate {
+    /// The split used when identifying changes.
+    ///
+    /// Common splits include:
+    /// - `""` for char-level.
+    /// - `" "` for word-level.
+    /// - `"\n" for line-level.
+    ///
+    /// Default: `"\n"`
+    pub fn split<S>(mut self, split: S) -> Self
+    where
+        S: Into<String>,
+    {
+        self.split = split.into();
+        self
+    }
+
+    /// The maximum allowed edit distance.
+    ///
+    /// Default: `0`
+    pub fn distance(mut self, distance: i32) -> Self {
+        self.distance = distance;
+        self
+    }
+}
+
+impl Predicate for DifferencePredicate {
+    type Item = str;
+
+    fn eval(&self, edit: &str) -> bool {
+        let change = difference::Changeset::new(&self.orig, edit, &self.split);
+        change.distance <= self.distance
+    }
+}
+
+/// Creates a new `Predicate` that diffs two strings.
+///
+/// # Examples
+///
+/// ```
+/// use predicates::predicate::*;
+///
+/// let predicate_fn = str::diff("Hello World");
+/// assert_eq!(true, predicate_fn.eval("Hello World"));
+/// ```
+pub fn diff<S>(orig: S) -> DifferencePredicate
```

Comment:
The behavior of this predicate seems backwards to me based on the name. `diff` to me sounds like the predicate should be satisfied (return true) when the strings _are different_. But the behavior seems to be that the result is true when the strings are similar. Is there a better name that could be used for this - maybe like `distance`?

Reply:
While this has features for edit-distance, as a user, my real care about is showing the diff between the strings once #7 is available and I was going for a name that communicated that.

I can see how the name implies the opposite meaning.  What if instead I reversed the behavior of `eval` instead?  The downside is that almost everyone will now need to `not` it.

Reply:
Would it make sense to create two different predicates that handle the two different cases (looking to see if strings differ, looking to see if strings are the same)? The underlying type could easily be shared between two facades with clear naming.

Reply:
Eh, sure.  Updated.